File: styleable-views/src/main/java/adapter/StyleableTextWatcherAdapter.java
package adapter;

import android.text.Editable;
import android.text.TextWatcher;

import view.StyleableEditText;

/**
 * Created by Chavit Thawanapong on 4/25/2017 AD.
 * Email: c.thawanapong@gmail.com
 */

public class StyleableTextWatcherAdapter implements TextWatcher {
    private static final String TAG = StyleableTextWatcherAdapter.class.getSimpleName();

    private final StyleableEditText view;
    private final StyleableTextWatcherListener listener;

    public StyleableTextWatcherAdapter(StyleableEditText editText, StyleableTextWatcherListener listener) {
        this.view = editText;
        this.listener = listener;
    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        listener.onTextChanged(view, s.toString());
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void afterTextChanged(Editable s) {

    }

    public interface StyleableTextWatcherListener {
        void onTextChanged(StyleableEditText view, String text);
    }
}

File: styleable-views/src/main/java/com/github/cthawanapong/adapter/StyleableTextWatcherAdapter.kt
package com.github.cthawanapong.adapter

import android.text.Editable
import android.text.TextWatcher
import com.github.cthawanapong.view.StyleableClearableEditText

/**
 * Created by CThawanapong on 27/1/2018 AD.
 * Email: c.thawanapong@gmail.com
 */
class StyleableTextWatcherAdapter(
        private val styleableClearableEditText: StyleableClearableEditText,
        private val listener: StyleableTextWatcherListener
) : TextWatcher {
    companion object {
        @JvmStatic
        private val TAG = StyleableTextWatcherAdapter::class.java.simpleName
    }

    override fun afterTextChanged(s: Editable?) {

    }

    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {

    }

    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
        listener.onTextChanged(styleableClearableEditText, s.toString())
    }

    interface StyleableTextWatcherListener {
        fun onTextChanged(view: StyleableClearableEditText, text: String)
    }
}